### PR TITLE
Permissions backend postgresql

### PR DESCRIPTION
**Pull request by leplatrem**
@Natim r?

### CODE REVIEW

File: cliquet/permission/memory.py [hunk 1]
```diff
@@ -76,10 +76,10 @@ def object_permission_authorized_principals(self, object_id, permission,
             keys = [(object_id, permission)]
         else:
             keys = get_bound_permissions(object_id, permission)
-        permissions = set([])
+        principals = set([])
         for obj_id, perm in keys:
-            permissions |= self.object_permission_principals(obj_id, perm)
-        return permissions
+            principals |= self.object_permission_principals(obj_id, perm)
+        return principals
```
Comment (Natim) [inline cliquet/permission/memory.py]: good catch

File: cliquet/permission/postgresql/__init__.py [hunk 2]
```diff
@@ -0,0 +1,178 @@
...
+        here = os.path.abspath(os.path.dirname(__file__))
+        schema = open(os.path.join(here, 'schema.sql')).read()
+        with self.connect() as cursor:
+            cursor.execute(schema)
+        logger.info('Created PostgreSQL permission tables')
+
+    def flush(self):
+        query = """
+        DELETE FROM user_principals;
+        DELETE FROM aces;
```
Comment (Natim) [inline cliquet/permission/postgresql/__init__.py]: Isn't ``access_control_entries`` more explicit?

File: cliquet/permission/postgresql/__init__.py [hunk 3]
```diff
@@ -0,0 +1,178 @@
...
+        )
+        SELECT principal
+          FROM required_perms JOIN aces
+            ON (object_id = column1 AND permission = column2)
+        ;""" % perms
+        with self.connect() as cursor:
+            cursor.execute(query)
+            results = cursor.fetchall()
+        return set([r['principal'] for r in results])
+
```
Comment (Natim) [inline cliquet/permission/postgresql/__init__.py]: We should had the check_permission method on postgresql ends as well here.

Comment (leplatrem) [inline cliquet/permission/postgresql/__init__.py]: postgresql ends ?

File: cliquet/permission/postgresql/__init__.py (hunk 2 above)
Comment (leplatrem) [inline cliquet/permission/postgresql/__init__.py]: oh yes!

File: cliquet/permission/postgresql/__init__.py (hunk 3 above)
Comment (Natim) [inline cliquet/permission/postgresql/__init__.py]: Build a SQL query that return ``True`` or ``False`` for the ``check_permission``

Comment (leplatrem) [inline cliquet/permission/postgresql/__init__.py]: Ah, ok. I just read the code of ``BasePermission.check_permission``.

But then why do we need ``object_permission_authorized_principals`` ?

Comment (Natim) [inline cliquet/permission/postgresql/__init__.py]: The simple answer is "for debugging"

Comment (leplatrem) [inline cliquet/permission/postgresql/__init__.py]: Then we can get rid of it :)

**Pull request merged**